src/components/QuickSearch/util.tsx

```diff
@@ -0,0 +1,60 @@
+import { isDefined } from '~/utils/type-guards';
+
+type MatchedFilter = {
+  indexName: string;
+  attribute: string;
+  attributeRegexp: RegExp;
+  matchRegexp: RegExp;
+  matches: string[] | null;
+};
+
+const applyQueryMatchers = (query: string) => {
+  const filters = [
+    {
+      indexName: 'models',
+      attribute: 'type',
+      attributeRegexp: /t:(\w+)/,
+      matchRegexp: /(?<=t:)\w+/,
+      matches: [],
+    },
+  ];
+
+  const matchedFilters: Array<MatchedFilter> = filters.map((filter) => {
+    const { matchRegexp } = filter;
+    return {
+      ...filter,
+      matches: query.match(matchRegexp),
+    };
+  });
+
+  const updatedQuery = filters
+    .reduce((acc, filter) => {
+      const { attributeRegexp } = filter;
+      return acc.replace(attributeRegexp, '');
+    }, query)
+    .trim();
+
+  return {
+    updatedQuery,
+    matchedFilters,
+  };
+};
+
+const getMatchedFiltersByIndexName = (indexName: string, matchedFilters: Array<MatchedFilter>) => {
```

Comment:
And here we can add the proper types
```suggestion
const getMatchedFiltersByIndexName = (indexName: FilterIndex, matchedFilters: MatchedFilter[]) => {
```
Just changed the array type to be consistent with how we do it in all other places